src/window.rs

```diff
@@ -209,7 +209,7 @@ impl Window {
             .with_visibility(false)
             .with_transparency(true)
             .with_decorations(window_config.decorations());
-        let window_builder = Window::platform_builder_ext(window_builder);
+        let window_builder = Window::platform_builder_ext(window_builder, title);
```

Comment:
Well, at least on Linux, the `--title` flag seems pretty useless, since running a command will change the title to the process name anyway.

Reply:
Agreed, maybe the `--title` flag should disable the program name title updates. But either way I think the behavior of `--class` is static.

Reply:
The `--title` flag has always set the window class not title. The `title` variable here is exactly what can be overwritten using the `--title` command line flag. What exactly do you mean by 'full solution' beyond what is currently happening @nixpulvis?

Reply:
wait... `--title` sets the class? That seems very confusing. So I guess the "full solution" I see now is:

1. Re-purpose `--title` flag` to actual set the title, and prevent updates to running program names.
2. Create a `--class` flag that does what `--title` does today.

This will break peoples configs who are depending on this behavior, but this is weird enough (at least to me) to be worth it.

Reply:
Well, the configs are already broken, because of changes to `winit`'s behaviour. And `--title` is also how termite does it, so I thought it would be fitting.

Also, [here](https://tronche.com/gui/x/xlib/ICC/client-to-window-manager/wm-class.html) it says that the first part of `WM_CLASS` is the application **name**, so if we don't use `--title`, we should use something like `--name` or `--classname`

Reply:
@Tezkerek how termite does what? I'm pretty sure what I described is how termite works. I have no idea where the notion of a title came from however, that link makes it seem, indeed, like we should only have a `--name` and `--class` flag, which I'm just as happy with.

Reply:
Yeah, I misremembered, you're right.

Reply:
I'm not entirely sure what the original design idea behind this was since I haven't written it. But setting both title and class differently would be fine.

However, setting the name doesn't make much sense. I don't think the application name is used in a lot of places and instead we should differentiate between title and class. Only the class itself is mentioned in the URL I (and @Tezkerek) have linked. The title uses `WM_NAME` and is currently already properly set afaik.

So the current `--title` option sets only the title (displayed in the bar above the window in some WMs). In this PR this is already set as window class, it would be possible to change this to use `--class` instead. Then `--title` would only have an effect if dynamic window title is disabled in the config (which seems fine).

Reply:
If I was understanding this correctly, the *name* is the same as the *title*. And *class* is the attribute that should be never changing (although possible to set via `--class`). The *title* Would then be (by default) the current running program, and possible to lock to a single value with `--title`.

Reply:
Ok I also struggled with the [linked](https://tronche.com/gui/x/xlib/ICC/client-to-window-manager/wm-class.html) docs. But your [comment](https://github.com/jwilm/alacritty/issues/1350#issuecomment-394492246) cleared it up for me.

I think we're all saying the same thing at this point.